### ISSUE TITLE
Unhelpful error when using named arguments on macro methods that do not support them

### ISSUE THREAD

**Issue by Blacksmoke16** [crystal-lang/crystal#15310]
This was brought up on the Discord, where the reduced code was something like:

https://play.crystal-lang.org/#/r/hj72

```cr
macro test(node)
  {% pp node.type.name generic_args: false %}
end
 
test name : Array(Int32)
```
```cr
In test.cr:5:1

 5 | test name : Array(Int32)
     ^---
Error: expanding macro


In test.cr:2:19

 2 | {% pp node.type.name generic_args: false %}
                     ^---
Error: named arguments are not allowed here


In test.cr:5:13

 5 | test name : Array(Int32)
                 ^
Error: named arguments are not allowed here
```
The trace points you to the right place, but the error could be a bit better. Maybe something along the lines of:

> 'Generic#name' does not allow named arguments

to make it more clear what specific overload does not support them.

**Comment by hahwul** [comment]
@Blacksmoke16 
Hi! I'd like to work on this good first issue.

Plan
- Improve the error message when named args are used on macros that don't support them.
- Make it say something like: "Macro 'test' does not support named arguments" (or reference the specific overload/macro name)
- Add a spec test for this case.

Is this still needed? Any preferred wording or additional info to include in the error? Thanks!

**Comment by Blacksmoke16** [comment]
@hahwul Yes that sounds good. I think the error message from the OP would be most ideal:

> 'Generic#name' does not allow named arguments

Where we include the ASTNode type and the method name to make it clear _what_ doesn't support named arguments.

EDIT: Or maybe just this to be more similar to the non-macro context:

> no parameter named 'generic_args'

Because thats the crux of the problem, can still provide positional arguments as named arguments, but the problem here is it just doesn't have any parameters at all
